src/test_node.js

```diff
@@ -0,0 +1,124 @@
+// @ts-check
+var _ = require('lodash');
+
+class TestNode {
+  /**
+   * @param {import('./factory').Factory} factory
+   * @param {string} nodeName
+   */
+  constructor(factory, nodeName) {
+    this._factory = factory;
+    this._nodeName = nodeName;
+    this._attributes = {};
+    this._properties = [];
+    this._children = [];
+  }
+
+  /**
+   * @param {string} name
+   * @param {string} value
+   * @returns {TestNode}
+   * @chainable
+   */
+  property(name, value) {
+    this._properties.push({ name: name, value: value });
+    return this;
+  }
+
+  /**
+   * @param {string} className
+   * @returns {TestNode}
+   * @chainable
+   */
+  className(className) {
+    this._attributes.classname = className;
+    return this;
+  }
+
+  /**
+   * @param {string} name
+   * @returns {TestNode}
+   * @chainable
+   */
+  name(name) {
+    this._attributes.name = name;
+    return this;
+  }
+
+  /**
+   * @param {string} timeInSeconds
+   * @returns {TestNode}
+   * @chainable
+   */
+  time(timeInSeconds) {
+    this._attributes.time = timeInSeconds;
+    return this;
+  }
+
+  /**
+   * @protected
+   * @param {import('xmlbuilder').XMLElement} parentElement
+   * @returns {import('xmlbuilder').XMLElement}
+   */
+  createNode(parentElement) {
+    return parentElement.ele(this._nodeName, this._attributes);
+  }
+
+  /**
+   * @protected
+   * @param {import('xmlbuilder').XMLElement} element
+   * @returns {import('xmlbuilder').XMLElement}
+   */
+  buildNode(element) {
+    if (this._properties.length) {
+      var propertiesElement = element.ele('properties');
+      _.forEach(this._properties, function (property) {
+        propertiesElement.ele('property', {
+          name: property.name,
+          value: property.value,
+        });
+      });
+    }
+    _.forEach(this._children, function (child) {
+      child.build(element);
+    });
+    return element;
+  }
+
+  /**
+   * @returns {number}
+   */
+  getTestCaseCount() {
+    throw new Error('Not implemented');
+  }
+
+  /**
+   * @returns {number}
+   */
+  getFailureCount() {
+    return 0;
+  }
+
+  /**
+   * @returns {number}
+   */
+  getErrorCount() {
+    return 0;
+  }
+
+  /**
+   * @returns {number}
+   */
+  getSkippedCount() {
+    return 0;
+  }
```

Comment:
Yea, should never be used. I was thinking that if the extending class doesn't implement it then returning 0 was fine rather than erroring 🤔 

Ideally this would be defined in the interface and use `implements Countable` or something like that so raising errors are probably closer to that.